core/src/main/java/com/sequenceiq/cloudbreak/core/bootstrap/service/SimpleSecurityService.java

```
@@ -0,0 +1,179 @@
+
+    @Value("${cb.cert.dir:" + CB_CERT_DIR + "}")
+    private String certDir;
+
+    @Value("${cb.tls.cert.file:" + CB_TLS_CERT_FILE + "}")
+    private String clientCert;
+
+    @Value("${cb.tls.private.key.file:" + CB_TLS_PRIVATE_KEY_FILE + "}")
+    private String clientPrivateKey;
+
+    @Inject
+    private StackRepository stackRepository;
+
+    public String getCertDir(Long stackId) {
+        return Paths.get(certDir + "/stack-" + stackId).toString();
+    }
+
+    public String prepareCertDir(Long stackId) throws CloudbreakSecuritySetupException {
+        LOGGER.info("Creating directory for the keys and certificates under {}", certDir);
+        Path stackCertDir = Paths.get(getCertDir(stackId));
+        if (!Files.exists(stackCertDir)) {
+            try {
+                Files.createDirectories(stackCertDir);
+                prepareFiles(stackId);
+            } catch (IOException | SecurityException se) {
+                throw new CloudbreakSecuritySetupException("Failed to create directory: " + stackCertDir);
+            }
+        } else {
+            prepareFiles(stackId);
+        }
+        return stackCertDir.toString();
+    }
+
+    private void prepareFiles(Long stackId) {
+        try {
+            Stack stack = stackRepository.findOne(stackId);
+            if (stack != null) {
+                Certificate certificate = stack.getCertificate();
+                if (certificate != null) {
+                    readServerCert(stack.getId(), stack.getCertificate());
+                    readClientCert(stack.getId(), stack.getCertificate());
+                    readClientKey(stack.getId(), stack.getCertificate());
+                    readPrivateSshKey(stack.getId(), stack.getCertificate());
+                    readPublicSshKey(stack.getId(), stack.getCertificate());
+                }
+            }
+        } catch (Exception ex) {
+            LOGGER.warn("Error occured when keys tried to repaired.");
```

Shouldn't we do something here instead of swallowing the exception message completely and returning successfully?